common/src/main/java/com/wynntils/wc/utils/scoreboard/ScoreboardManager.java

```diff
@@ -67,30 +68,37 @@ public final class ScoreboardManager {
         }
 
         List<ScoreboardLine> scoreboardCopy = new ArrayList<>(reconstructedScoreboard);
+        LinkedList<ScoreboardLineChange> queueCopy = new LinkedList<>(queuedChanges);
 
-        for (ScoreboardLineChange queuedChange : queuedChanges) {
-            if (queuedChange.method() == ServerScoreboard.Method.REMOVE) {
-                scoreboardCopy.removeIf(
-                        scoreboardLine -> Objects.equals(scoreboardLine.line(), queuedChange.lineText()));
-            }
+        Map<Integer, ScoreboardLine> scoreboardLineMap = new TreeMap<>();
+
+        for (ScoreboardLine scoreboardLine : scoreboardCopy) {
+            scoreboardLineMap.put(scoreboardLine.index(), scoreboardLine);
         }
 
         Set<String> changedLines = new HashSet<>();
-        while (!queuedChanges.isEmpty()) {
-            ScoreboardLineChange processed = queuedChanges.pop();
+        while (!queueCopy.isEmpty()) {
+            ScoreboardLineChange processed = queueCopy.pop();
 
             if (processed.method() == ServerScoreboard.Method.REMOVE) {
+                for (ScoreboardLine lineToRemove : scoreboardLineMap.values().stream()
```

Comment:
Also, this could be lambdas all way, right? Skip the for and toList and to the remove in a forEach() instead.

Reply:
forEach does not work there for some reason. It is 2 am so I am either insane or I am missing something (It also didn't work now and before commiting this). For should be fine \:).

Reply:
That's odd. It looks like it should work just fine. But ok, send it in as-is, and I can play around with it later.